From 09513337737aaa2822a17882ad4849e5c5497b5a Mon Sep 17 00:00:00 2001
From: Tianshu Qiu <tian.shu.qiu@intel.com>
Date: Thu, 16 Apr 2015 16:10:32 +0800
Subject: [PATCH 360/360] atomisp2: Revise dvs coefficents setting

This patch fixes the setting of dvs coefficients
by calling atomisp_css_cp_dvs2_coefs().

Change-Id: I314f86ea4e21802f124aa7674042fac07480bf7f
Fixes-Issue: #JIMINAN-31158
Signed-off-by: Tianshu Qiu <tian.shu.qiu@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/51556
Approver: Bin B Han <bin.b.han@intel.com>
Reviewed-by: Bin B Han <bin.b.han@intel.com>
Integrator: Wei Tao <wei.tao@intel.com>
Validation-Android: Wei Tao <wei.tao@intel.com>
Build: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
Maintainer: Wei Tao <wei.tao@intel.com>
---
 .../drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c       | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c
index 16351ea..2a15c9e 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c
@@ -2762,8 +2762,12 @@ static long atomisp_vidioc_default(struct file *file, void *fh,
 	case ATOMISP_IOC_G_DVS2_BQ_RESOLUTIONS:
 		err = atomisp_get_dvs2_bq_resolutions(asd, arg);
 		break;
+
 	case ATOMISP_IOC_S_DIS_COEFS:
-		err = atomisp_set_dis_coefs(asd, arg);
+		err = atomisp_css_cp_dvs2_coefs(asd, arg,
+				&asd->params.css_param, true);
+		if (!err && arg)
+			asd->params.css_update_params_needed = true;
 		break;
 
 	case ATOMISP_IOC_S_DIS_VECTOR:
-- 
1.9.1

